### PR TITLE
[MOBILE-1652] Add analytics methods and properties

### PR DESCRIPTION
**Pull request by rlepinski**
### What do these changes do?
Adds screen tracking and data collection flags.

### How did you verify these changes?
Manually in the sample app for both iOS and Android.

### CODE REVIEW

File: documentation/index.md [hunk 1]
```diff
@@ -80,6 +79,24 @@ Sets the namedUser for the device.
     Airship.namedUser = "totes mcgoats"
 ```
 
+### isDataCollectionEnabled
+
+Enables or disables data collection. To require data be opted in before collected, set
+`com.urbanairship.data_collection_opt_in_enabled` to true in the app's config. Disabling
+data collection disables tags, attributes, analytics, named user, and push. To allow the user to still receive broadcast push, set `isPushTokenRegistrationEnabled` to true.
```
Comment (BrianBatchelder) [inline documentation/index.md]:
```suggestion
data collection disables tags, attributes, analytics, named user, and push. To allow the user to still receive broadcast pushes, set `isPushTokenRegistrationEnabled` to true.
```

File: documentation/index.md [hunk 2]
```diff
@@ -80,6 +79,24 @@ Sets the namedUser for the device.
...
+
+```
+    Airship.isDataCollectionEnabled = true;
+```
+
+### isPushTokenRegistrationEnabled
+
+Enables or disables push token registration. This value defaults to the current value of `isDataCollectionEnabled`. Can be used to enable push and still have data collection disabled.
```
Comment (BrianBatchelder) [inline documentation/index.md]:
```suggestion
Enables or disables push token registration. This value defaults to the current value of `isDataCollectionEnabled`. Can be used to enable broadcast push and still have data collection disabled.
```

File: documentation/index.md [hunk 3]
```diff
@@ -96,6 +113,7 @@ The badge number on iOS.
     Airship.badgeNumber = 3
 ```
 
+
```
Comment (BrianBatchelder) [inline documentation/index.md]:
```suggestion
```

File: documentation/index.md [hunk 4]
```diff
@@ -231,6 +249,15 @@ Adds a custom event.
...
 
+### trackScreen(screenName)
+
+Screen tracking event.
+
+```
+    Airship.trackScreen("home")
+```
+
+
```
Comment (BrianBatchelder) [inline documentation/index.md]:
```suggestion
```

**Review by BrianBatchelder — APPROVED** [review]
Minor stuff. Pre-approved.
👍

File: android/src/ti/airship/AirshipTitaniumModule.java [hunk 5]
```diff
@@ -100,6 +99,35 @@ public void setNamedUser(String namedUser) {
...
+
+    @Kroll.method
+    @Kroll.setProperty
+    public void setIsDataCollectionEnabled(boolean enabled) {
+        UAirship.shared().setDataCollectionEnabled(enabled);
+    }
+
+    @Kroll.getProperty
+    @Kroll.method
+    public boolean getIsDataCollectionEnabled() {
```
Comment (marc-scig) [inline android/src/ti/airship/AirshipTitaniumModule.java:115]: should these methods be named `setDataCollectionEnabled` and `getDataCollectionEnabled`? that would be more in line with our other JS ecosystem stuff.

**Review by marc-scig — COMMENTED** [review]
Looks good but I have a suggestion re: names

File: android/src/ti/airship/AirshipTitaniumModule.java (hunk 5 above)
Comment (rlepinski) [reply android/src/ti/airship/AirshipTitaniumModule.java:115]: these are properties while the other ones are methods:
```
Airship.isDataCollectionEnabled = true
```

Comment (marc-scig) [reply android/src/ti/airship/AirshipTitaniumModule.java:115]: oh, right

**Review by marc-scig — APPROVED** [review]

**Pull request merged**